File: publicApi .NET/publicApi/publicApi/Models/Currency .cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace publicApi.Models
{
    public class Currency
    {
        public int currencyId { get; set; }
        public string title { get; set; }
        public string currency { get; set; }
        public float value { get; set; }
        public string data { get; set; }
    }
}

/*
  const CurrencySchema = mongoose.Schema({
  currencyId: {
    type: Number,
    required: true,
    index: { unique: true }
  },  
  title: String,
  currency: String,
  value: Number,
  date: String
});
*/

File: database NodeJS/Models/currency.js
const mongoose = require("mongoose");
const { CurrencyDTO } = require("../DTOs/CurrencyDTO")

const CurrencySchema = mongoose.Schema({
  currencyId: {
    type: Number,
    required: true,
    index: { unique: true }
  },  
  title: String,
  currency: String,
  value: Number,
  date: String
});

CurrencySchema.statics.toDTO = function (modelObject) {
  return new CurrencyDTO(
      modelObject.currencyId,
      modelObject.title,
      modelObject.currency,
      modelObject.value,
      modelObject.date)
}
//CurrencySchema.plugin(uniqueValidator);


const Currency = mongoose.model('Currency', CurrencySchema);

module.exports = { Currency : Currency };
